Actions/IncrementVersionNumber/IncrementVersionNumber.psm1

```diff
@@ -80,25 +81,45 @@ function Set-VersionInSettingsFile {
             # Increment major version number
             $versionNumbers += $oldVersion.Major + 1
             $versionNumbers += 0
+            # Include build number if it exists in the old version number
+            if ($oldVersion.Build -ne -1) {
+                $versionNumbers += 0
+            }
         }
         '+0.1' {
             # Increment minor version number
             $versionNumbers += $oldVersion.Major
             $versionNumbers += $oldVersion.Minor + 1
-
+            # Include build number if it exists in the old version number
+            if ($oldVersion.Build -ne -1) {
+                $versionNumbers += 0
+            }
+        }
+        '+0.0.1' {
+            # Increment minor version number
+            $versionNumbers += $oldVersion.Major
+            $versionNumbers += $oldVersion.Minor
+            if ($oldVersion.Build -eq -1) {
+                $versionNumbers += 1
+            }
+            else {
+                $versionNumbers += $oldVersion.Build + 1
+            }
         }
         default {
+            Write-Host "newValue: $newValue"
+            Write-Host ($oldVersion -eq $null)
```

Comment:
These kinds of outputs are useful when debugging, but otherwise clutter the logs a lot.